x-pack/plugins/ml/server/shared_services/providers/system.ts

```diff
@@ -14,73 +15,79 @@ import { capabilitiesProvider } from '../../lib/capabilities';
 import { MlInfoResponse } from '../../../common/types/ml_server_info';
 import { ML_RESULTS_INDEX_PATTERN } from '../../../common/constants/index_patterns';
 import { MlCapabilitiesResponse, ResolveMlCapabilities } from '../../../common/types/capabilities';
-import { SharedServicesChecks } from '../shared_services';
+import { GetGuards } from '../shared_services';
 
 export interface MlSystemProvider {
   mlSystemProvider(
-    mlClusterClient: ILegacyScopedClusterClient,
     request: KibanaRequest
   ): {
     mlCapabilities(): Promise<MlCapabilitiesResponse>;
     mlInfo(): Promise<MlInfoResponse>;
-    mlAnomalySearch<T>(searchParams: SearchParams): Promise<SearchResponse<T>>;
+    mlAnomalySearch<T>(searchParams: RequestParams.Search<any>): Promise<SearchResponse<T>>;
   };
 }
 
 export function getMlSystemProvider(
-  { isMinimumLicense, isFullLicense, getHasMlCapabilities }: SharedServicesChecks,
+  getGuards: GetGuards,
   mlLicense: MlServerLicense,
   spaces: SpacesPluginSetup | undefined,
   cloud: CloudSetup | undefined,
   resolveMlCapabilities: ResolveMlCapabilities
 ): MlSystemProvider {
   return {
-    mlSystemProvider(mlClusterClient: ILegacyScopedClusterClient, request: KibanaRequest) {
-      // const hasMlCapabilities = getHasMlCapabilities(request);
-      const { callAsInternalUser } = mlClusterClient;
+    mlSystemProvider(request: KibanaRequest) {
       return {
         async mlCapabilities() {
-          isMinimumLicense();
+          return await getGuards(request)
+            .isMinimumLicense()
+            .ok(async ({ scopedClient }) => {
+              const { isMlEnabledInSpace } =
+                spaces !== undefined
+                  ? spacesUtilsProvider(spaces, request)
+                  : { isMlEnabledInSpace: async () => true };
 
-          const { isMlEnabledInSpace } =
-            spaces !== undefined
-              ? spacesUtilsProvider(spaces, request)
-              : { isMlEnabledInSpace: async () => true };
+              const mlCapabilities = await resolveMlCapabilities(request);
+              if (mlCapabilities === null) {
+                throw new Error('mlCapabilities is not defined');
```

Comment:
"* are not defined" ?

Reply:
i believe this is correct being singular, as it is a single collection of settings that isn't defined.